components/browser/session/src/main/java/mozilla/components/browser/session/LegacySessionManager.kt

```diff
@@ -273,20 +273,21 @@ class LegacySessionManager(
         getEngineSession(session)?.let { return it }
 
         return engine.createSession(session.private).apply {
+            var restored = false
             session.engineSessionHolder.engineSessionState?.let { state ->
-                restoreState(state)
+                restored = restoreState(state)
                 session.engineSessionHolder.engineSessionState = null
             }
 
-            link(session, this)
+            link(session, this, restored)
```

Comment:
Ah, interesting, you return whether the restore was successful or not. Initially I thought the engine session you keep a flag internally after restore() was called (meaning "I am a session that was restored") which we then could use to decide whether we need to call `load()`. But I think your implementation makes more sense!